proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -418,6 +419,8 @@ private void sendMessage(ChatMessageType position, String message)
     @Override
     public void sendMessage(ChatMessageType position, BaseComponent... message)
     {
+        // transform score components
```

Comment:
I'm not sure mutating a message the plugin has used is a good idea because it might cause unwanted change. I think it should be left for the plugin author to implement if wanted.

Reply:
The only way to successfully mutate a ScoreComponent is to populate the BungeeCord Scoreboard with values. Besides that, from what I could see, the Scoreboard system currently implemented in BungeeCord does not look very polished. So I don't really see where the unwanted change would come from  unless the user specifically populates the BungeeCord scoreboard with the values they want to use; in which case it'd probably be better to just use a TextComponent.

Reply:
Bungee populates the scoreboard with the values of the server they are currently on btw

Reply:
I see.

I think the Scoreboard argument should be added back in ChatComponentSerializer#transformScoreComponent.

Here's my reasoning:
A server is a game mode with two teams (red, blue). You want to show blue player a score of a red player.

If a plugin decides they want to do this, they can then call this method manually attaching the desired scoreboard on their ScoreComponent

This looks like it wouldn't cause any issues as the automatic transform in sendMessage wouldn't do anything because the value is already input.

Reply:
A bit late sorry, but you could just pass in the other ProxiedPlayer and it will use their Scoreboard.